modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java

```diff
@@ -380,7 +381,12 @@ public void processOpts() {
         } else if (JERSEY2.equals(getLibrary())) {
             supportingFiles.add(new SupportingFile("JSON.mustache", invokerFolder, "JSON.java"));
             supportingFiles.add(new SupportingFile("ApiResponse.mustache", invokerFolder, "ApiResponse.java"));
-            supportingFiles.add(new SupportingFile("auth/HttpSignatureAuth.mustache", authFolder, "HttpSignatureAuth.java"));
+            Map<String, SecurityScheme> securitySchemeMap = openAPI != null ?
+                    (openAPI.getComponents() != null ? openAPI.getComponents().getSecuritySchemes() : null) : null;
+            List<CodegenSecurity> authMethods = fromSecurity(securitySchemeMap);
+            if (ProcessUtils.hasHttpSignatureMethods(authMethods)) {
```

Comment:
What about using `ProcessUtils.hasHttpSignatureMethods(openAPI)` instead and remove line 384 to 386?

Reply:
done.